tools/bcr_validation.py

```diff
@@ -226,27 +212,6 @@ def verify_module_dot_bazel(self, module_name, version):
               self.report(BcrValidationResult.FAILED, f"The patch file `{patch_file}` has expected integrity value `{expected_integrity}`, but the real integrity value is `{actual_integrity}`.")
             apply_patch(source_root, source["patch_strip"], str(patch_file.resolve()))
 
-    source_module_dot_bazel = source_root.joinpath("MODULE.bazel")
-    if source_module_dot_bazel.exists():
-      source_module_dot_bazel_content = open(source_module_dot_bazel, "r").readlines()
-    else:
-      source_module_dot_bazel_content = []
-    bcr_module_dot_bazel_content = open(self.registry.get_module_dot_bazel_path(module_name, version), "r").readlines()
-    source_module_dot_bazel_content = fix_line_endings(source_module_dot_bazel_content)
-    bcr_module_dot_bazel_content = fix_line_endings(bcr_module_dot_bazel_content)
-    file_name = "a/" * int(source.get("patch_strip", 0)) + "MODULE.bazel"
-    diff = list(unified_diff(source_module_dot_bazel_content, bcr_module_dot_bazel_content, fromfile=file_name, tofile=file_name))
-
-    if diff:
-      self.report(BcrValidationResult.FAILED,
-                  "Checked in MODULE.bazel file doesn't match the one in the extracted and patched sources.\n"
-                  + f"Please fix the MODULE.bazel file or you can add the following patch to {module_name}@{version}:\n"
-                  + "    " + "    ".join(diff))
-      if self.should_fix:
-        self.add_module_dot_bazel_patch(diff, module_name, version)
-    else:
-      self.report(BcrValidationResult.GOOD, "Checked in MODULE.bazel matches the sources.")
```

Comment:
If we expect all new versions to include overlay -- I think it's simpler to remove this functionality entirely.
We don't need to support it for older versions as they are already "validated"

Reply:
I don't think it's reasonable to assume that all new versions will only support Bazel 7.2.0+.